xbmc/addons/GUIDialogAddonInfo.cpp

```diff
@@ -460,6 +508,85 @@ void CGUIDialogAddonInfo::OnSettings()
   CGUIDialogAddonSettings::ShowForAddon(m_localAddon);
 }
 
+//! \brief Recursively add add-on dependencies to select dialog
+static void FillSelectDialogWithAddons(std::map<std::string, std::string>& added,
+                                       CGUIDialogSelect* pDialog,
+                                       const std::string& id)
+{
+  AddonPtr addon;
+  if (CServiceBroker::GetAddonMgr().FindInstallableById(id, addon))
+  {
+    for (const auto& it : addon->GetDeps())
+    {
+      if (StringUtils::StartsWith(it.first, "xbmc.") ||
+          StringUtils::StartsWith(it.first, "kodi."))
+        continue;
+      if (added.find(it.first) != added.end())
+      {
+        if (!added[it.first].empty() &&
+            AddonVersion(it.second.first.asString()) < AddonVersion(added[it.first]))
+          continue;
+      }
+      added[it.first] = it.second.first.asString();
+      AddonPtr addon2;
+      CServiceBroker::GetAddonMgr().FindInstallableById(it.first, addon2);
+      if (addon2)
+      {
+        CFileItem item(addon2->Name());
+        std::stringstream str;
+        str << it.first << " " << it.second.first.asString();
+        AddonPtr ad;
+        CServiceBroker::GetAddonMgr().GetAddon(it.first, ad);
+        if (it.second.second || ad)
+        {
+          str << " (";
+          if (ad)
+            str << g_localizeStrings.Get(39019);
+          if (it.second.second)
+          {
+            if (ad)
+              str << ", ";
+            str << g_localizeStrings.Get(39018);
+          }
+          str << ")";
```

Comment:
done (though i doubt seeing (optional, installed) instead of (installed, optional) would kill any kittens).